internal/codeintel/dependencies/transport/graphql/lockfile_index_connection_resolver.go

```diff
@@ -0,0 +1,48 @@
+package graphql
+
+import (
+	"context"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend/graphqlutil"
+)
+
+type LockfileIndexConnectionResolver struct {
+	resolvers  []*LockfileIndexResolver
+	totalCount int
+	nextOffset *int
+}
+
+func NewLockfileIndexConnectionConnection(resolvers []*LockfileIndexResolver, totalCount int, nextOffset *int) *LockfileIndexConnectionResolver {
+	return &LockfileIndexConnectionResolver{
+		resolvers:  resolvers,
+		totalCount: totalCount,
+		nextOffset: nextOffset,
+	}
+}
+
+func (r *LockfileIndexConnectionResolver) Nodes(ctx context.Context) (resolvers []graphqlbackend.LockfileIndexResolver) {
+	resolvers = make([]graphqlbackend.LockfileIndexResolver, len(r.resolvers))
+	for i, r := range r.resolvers {
+		resolvers[i] = r
+	}
```

Comment:
```suggestion
	copy(resolvers, r.resolvers)
```

Is this just to prevent mutation at callsites?

Reply:
just to correct the type.